onebusaway-android/src/main/java/org/onebusaway/android/directions/util/LocationUtil.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2012 University of South Florida
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and 
+ * limitations under the License.
+ */
+
+package org.onebusaway.android.directions.util;
+
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.location.Address;
+import android.location.Geocoder;
+import android.location.Location;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.onebusaway.android.R;
+import org.onebusaway.android.io.elements.ObaRegion;
+import org.onebusaway.android.util.RegionUtils;
+
+
+/**
+ * Various utilities related to location data
+ *
+ * @author Khoa Tran
+ * @author Simon Jacobs (integration for onebusaway-android)
+ */
+
+
+public class LocationUtil {
+
+    private static final String TAG = "LocationUtil";
```

Comment:
This TAG needs to change, as it currently collides with the `LocationUtils` class.